lib/elixir/pages/Typespecs.md

```diff
@@ -25,65 +25,72 @@ The syntax Elixir provides for type specifications is similar to [the one in Erl
+          | maybe_improper_list(type1, type2)           # proper or improper list
+          | nonempty_improper_list(type1, type2)        # improper list
+          | nonempty_maybe_improper_list(type1, type2)  # non-empty proper or improper list
+
           | Literals                # Described in section "Literals"
-          | Builtin                 # Described in section "Built-in types"
+          | BuiltIn                 # Described in section "Built-in types"
           | Remotes                 # Described in section "Remote types"
 
 ### Literals
 
 The following literals are also supported in typespecs:
 
     type :: :atom                         ## Atoms
-          | 1.0                           ## Floats
-          | 1                             ## Integers
-          | 1..10                         ## Integers from 1 to 10
 
                                           ## Bitstrings
           | <<>>                          # empty bitstring
           | <<_::size>>                   # size is 0 or a positive integer
           | <<_::_ * unit>>               # unit is an integer from 1 to 256
-          | <<_::size, _::_*unit>>
+          | <<_::size, _::_ * unit>>
 
                                           ## Functions
           | (... -> type)                 # any arity, returns type
           | (() -> type)                  # 0-arity, returns type
           | (type1, type2 -> type)        # 2-arity, returns type
 
+                                          ## Integers
+          | 1                             # integer
+          | 1..10                         # integers from 1 to 10
+
                                           ## Lists
           | [type]                        # list with any number of type elements
           | []                            # empty list
           | [...]                         # shorthand for nonempty_list(any())
           | [type, ...]                   # shorthand for nonempty_list(type)
-          | [key: type]                   # keyword lists
+          | [key: value_type]             # keyword list with key is atom :key and value of value_type
 
-                                          ## Maps
-          | %{}                           # empty map
-          | %{...}                        # any map
-          | %{key: type}                  # map with required key :key with value of type
-          | %{required(type1) => type2}   # map with required keys of type1 with values of type2
-          | %{optional(type1) => type2}   # map with optional keys of type1 with values of type2
-          | %SomeStruct{}                 # struct with all fields of any type
-          | %SomeStruct{key: type}        # struct with :key field of type
+                                                  ## Maps
+          | %{}                                   # empty map
+          | %{key: value_type}                    # map with required key :key and value of value_type
```

Comment:
So here's the thing. previously we have `key:`, `type1`, `type2` and `type`,
I think by using `key:`, `key_type` and `value_type` people don't even need to read the comment, because it is pretty self explanatory,
I tried doing "map with required keys of key_type with values of value_type" but it was too long and would add a scroll bar, and in addition that is too verbose.
Could we find a shorter way to express it and still keep `key_type` and `value_type`?